extensions/nyctbustrips/nyctbustrips.go

```diff
@@ -0,0 +1,29 @@
+package nyctbustrips
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jamespfennell/gtfs/extensions"
+	gtfsrt "github.com/jamespfennell/gtfs/proto"
+)
+
+// Extension returns the NYCT bus trips extension
+func Extension() extensions.Extension {
+	return extension{}
+}
+
+type extension struct {
+	extensions.NoExtensionImpl
+}
+
+func (e extension) UpdateTrip(trip *gtfsrt.TripUpdate, feedCreatedAt uint64) extensions.UpdateTripResult {
+	tripId := trip.Trip.GetTripId()
+	vehicleId := trip.Vehicle.GetId()
+	direction := trip.Trip.GetDirectionId()
+	compositeTripId := strings.Replace(fmt.Sprintf("%s_%s_%d", tripId, vehicleId, direction), " ", "_", -1)
```

Comment:
On a related note, I think with this code the Trip ID may now vary across the trip? Like initially the trip won't have a vehicle assigned. Then when a vehicle is assigned, the trip ID will change. In Transiter this will result in any existing data for the trip being deleted when the ID change happens. I think this is totally fine (and we can always change it down the line) but just mentioning it anyway.

Reply:
This is a great point - I think tracking a trip end-to-end will be tricky with this feed's structure unfortunately 🙃.